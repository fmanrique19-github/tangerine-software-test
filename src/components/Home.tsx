import React from 'react';
import { connect } from 'react-redux';
import Header from  './Header';
import Footer from  './Footer';
import Albums from "./Albums";
import Photos from "./Photos";
import IAlbums from '../models/Albums';
import IUser from '../models/User';
import { IRootState } from '../reducers/RootReducer';
import { MDBContainer, MDBRow } from "mdbreact";
import IPhotos from '../models/Photos';
import runSaga from '../store/saga/injectSaga';
import { getPhotos } from '../sagas/PhotosSaga';

export interface IProps {
    user: IUser;
    albums: IAlbums;
    photos: IPhotos;
}

export interface IState {
    mdAlbums: number;
    mdPhotos: number;
    classPhotos: string;
    classAlbumsMobile: string;
    title: string;
    albumId: number;
}

class Home extends React.Component<IProps, IState> {

    constructor(props: IProps) {

        super(props);

        this.state = {
            mdAlbums: 12,
            mdPhotos: 0,
            classPhotos: 'd-none',
            classAlbumsMobile: '',
            title: '',
            albumId: 0,
        }
    }

    public displayPhotos = (albumId: number, title: string) => {

        if(this.state.mdPhotos === 0) {
            window.scrollTo(0, 0);
        }
        
        runSaga(getPhotos, [albumId]);
        
        this.setState({
            ...this.state,
            mdAlbums: 6,
            mdPhotos: 6,
            classPhotos: '',
            classAlbumsMobile: 'hide-albums-mobile',
            title: title,
            albumId: albumId,
        });
    }

    public displayAlbums = () => {
        this.setState({
            mdAlbums: 12,
            mdPhotos: 0,
            classPhotos: 'd-none',
            classAlbumsMobile: '',
        });
    }

    public render() {

        const { user, albums, photos } = this.props;

        return (
            user.isLogin ?
                <div>
                    <Header home />
                    <MDBContainer className="tangerine-container" fluid>
                        <MDBRow>
                            <Albums albums={albums} md={this.state.mdAlbums} displayPhotos={this.displayPhotos} classAlbumsMobile={this.state.classAlbumsMobile} />
                            <Photos photos={photos} md={this.state.mdPhotos} className={this.state.classPhotos} title={this.state.title} displayAlbums={this.displayAlbums} />
                        </MDBRow>
                    </MDBContainer>
                    <Footer />
                </div>
                : <div></div>
        );
    }
}

const mapStateToProps = (state: IRootState) => {
    return {
        user: state.user,
        albums: state.albums,
        photos: state.photos,
    };
};

export default connect(
    mapStateToProps,
    null,
)(Home);
