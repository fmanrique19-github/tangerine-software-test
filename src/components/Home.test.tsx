import React from 'react';
import renderer from 'react-test-renderer';
import Home from './Home';
import mockStore from '../tests/helpers/mockStore';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

jest.mock('../store/saga/injectSaga');

describe('Home Component', () => {

    it('Should display home page with no albums and no photos', () => {

        const props = {
            user: {
                isLogin: true,
            },
            albums: [],
            photos: [],
        };

        const tree = renderer
            .create(<Provider store={mockStore(props)}><Router><Home {...props} /></Router></Provider>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Should display div tag', () => {

        const props = {
            user: {
                isLogin: false,
            },
            albums: [],
            photos: [],
        };

        const tree = renderer
            .create(<Provider store={mockStore(props)}><Router><Home {...props} /></Router></Provider>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

});
