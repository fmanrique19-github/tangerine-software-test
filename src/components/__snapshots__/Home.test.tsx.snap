// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Should display div tag 1`] = `<div />`;

exports[`Home Component Should display home page with no albums and no photos 1`] = `
<div>
  <nav
    className="navbar-dark fixed-top orange darken-1 navbar navbar-expand-md pt-3"
    data-test="navbar"
    role="navigation"
  >
    <a
      aria-current={null}
      className="navbar-brand"
      data-test="navbar-brand"
      href="/home"
      onClick={[Function]}
    >
      <h2>
        <span
          className="float-left mr-2"
        >
          <svg
            fill="none"
            height="32"
            id="i-photo"
            stroke="currentcolor"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            viewBox="0 0 32 32"
            width="32"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 24 L12 16 2 26 2 2 30 2 30 24 M16 20 L22 14 30 22 30 30 2 30 2 24"
            />
            <circle
              cx="10"
              cy="9"
              r="3"
            />
          </svg>
        </span>
        Magic Photos
      </h2>
    </a>
    <button
      className="navbar-toggler"
      data-test="navbar-toggler"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      data-test="collapse"
      style={Object {}}
    >
      <ul
        className="navbar-nav mr-auto"
        data-test="navbar-nav"
      >
        <li
          className="nav-item active"
          data-test="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link Ripple-parent h5-responsive"
            data-test="nav-link"
            href="/home"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
          >
            Home
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </a>
        </li>
        <li
          className="nav-item"
          data-test="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link Ripple-parent h5-responsive"
            data-test="nav-link"
            href="/author"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
          >
            Author
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </a>
        </li>
      </ul>
      <ul
        className="navbar-nav ml-auto"
        data-test="navbar-nav"
      >
        <li
          className="nav-item"
          data-test="nav-item"
        >
          <a
            aria-current={null}
            className="nav-link Ripple-parent"
            data-test="nav-link"
            href="/logout"
            onClick={[Function]}
            onMouseUp={[Function]}
            onTouchStart={[Function]}
          >
            <h4>
              <strong>
                Admin
              </strong>
               (logout)
              <div
                className="float-right ml-2"
              >
                <svg
                  fill="none"
                  height="32"
                  id="i-signout"
                  stroke="currentcolor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 32 32"
                  width="32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M28 16 L8 16 M20 8 L28 16 20 24 M11 28 L3 28 3 4 11 4"
                  />
                </svg>
              </div>
            </h4>
            <div
              className="Ripple "
              data-test="waves"
              style={
                Object {
                  "height": "0px",
                  "left": "0px",
                  "top": "0px",
                  "width": "0px",
                }
              }
            />
          </a>
        </li>
      </ul>
    </div>
  </nav>
  <div
    className="container-fluid tangerine-container"
    data-test="container"
  >
    <div
      className="row"
      data-test="row"
    >
      <div
        className="col-md-12"
        data-test="col"
      >
        <div
          className="jumbotron"
          data-test="jumbotron"
        >
          <h1
            className="text-center"
          >
            There is no albums available
          </h1>
        </div>
      </div>
      <div
        className="col tangerine-fixed d-none"
        data-test="col"
      >
        <div
          className="jumbotron"
          data-test="jumbotron"
        >
          <h1
            className="text-center"
          >
            There is no photos available
          </h1>
        </div>
      </div>
    </div>
  </div>
  <footer
    className="page-footer blue font-small pt-4 mt-4"
    data-test="footer"
  >
    <div
      className="footer-copyright text-center py-3"
    >
      <div
        className="container-fluid"
        data-test="container"
      >
        Â© 
        2019
         Copyright - 
        <a
          className="text-white"
          href="http://www.tangerinesoftware.com"
          rel="noopener noreferrer"
          target="_blank"
        >
           tangerine 1.0
        </a>
      </div>
    </div>
  </footer>
</div>
`;
